### ISSUE TITLE
CRIT Service has terminated with an error

### ISSUE THREAD

**Issue by stanfieldr** [input-output-hk/jormungandr#1418]
**Describe the bug**

Sometimes jormungandr shuts down. Even if I have RUST_BACKTRACE=full, it does not provide any additional information. I can generally stay online for a few hours before the issue happens. I started seeing this issue in either 8.3 or 8.4.

**Mandatory Information**
1. jcli 0.8.3 (HEAD-8f276c00, release, linux [x86_64]) - [rustc 1.39.0 (4560ea788 2019-11-04)]
2. jormungandr 0.8.4 (HEAD-d443d0c5, release, linux [x86_64]) - [rustc 1.39.0 (4560ea788 2019-11-04)]

**To Reproduce**
1. Start jormungandr with pool settings (haven't tested passive)
2. Let it run for variable amount of time

**Expected behavior**
I expect jormungandr to run without error

**Additional context**
I redirect stdout to a log file and noticed it is outputting that to stderr instead.

stderr: A service has terminated with an error
stdout: attached

[pool_log.zip](https://github.com/input-output-hk/jormungandr/files/3989665/pool_log.zip)

**Comment by eugene-babichenko** [comment]
This line looks suspicious

> Dec 20 17:07:51.145 ERRO Error while accepting connection on 0.0.0.0:8299: Os { code: 24, kind: Other, message: "Too many open files" }, task: network

Can you please provide the output of `jcli rest v0 diagnostic get`?

**Comment by stanfieldr** [comment]
```
{"open_files_limit":1024,"cpu_usage_limit":18446744073709551615}
```

**Comment by fdk0** [comment]
I thought I share this. On macOS I get the node crashing running itn_rewards_v1 with genesis hash "8e4d2a343f3dcf9330ad9035b3e8d168e6728904262f2c434a4f8f934ec7b676" but running  rev0.8.5-alpha1 with genesis hash "9409af111b04896c756c1cee3b7f9bae8b9ed1843c9e0a5f07d92ab9b62f6f78" and nightly config works fine I had no crash since I started the node 4h ago, before the node crashed almost instantly.. not sure if it is relevant but I thought I should share it FYI. Any other  config since v0.8.0 resulted in a crash on macOS both Catalina and High Sierra. I run on an old MacBook 2009 with High Sierra and I got the "Too many open files" error which I solved setting "max_connections": 75 in my case. Sharing what I got let me know if I can help. Cheers!

**Comment by eugene-babichenko** [comment]
`max_connections` may be relevant here.

`open_files_limit` is a bit confusing. It is the limit for the total number of file descriptors per process. This includes not only files, but also open network connections, etc.

**Comment by stanfieldr** [comment]
I fixed the issue, thanks

**Comment by zmeel** [comment]
> I fixed the issue, thanks
I have the same error. How did you fix this?

**Comment by chadle-git** [comment]
Yea how did you fix?  I just got this on 8.15
